### PR TITLE
Fix/484

### LINKED ISSUE #484: [BUG] Apply on ovh_cloud_project_kube_nodepool crashes
### Terraform Version

```
❯ terraform -v
Terraform v1.6.2
on darwin_amd64
+ provider registry.terraform.io/ovh/ovh v0.34.0
```

### OVH Terraform Provider Version

```
❯ terraform init

Initializing the backend...

Initializing provider plugins...
- terraform.io/builtin/terraform is built in to Terraform
- Reusing previous version of ovh/ovh from the dependency lock file
- Using previously-installed ovh/ovh v0.34.0

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
```

### Affected Resource(s)

Please list the resources as a list, for example:
- ovh_cloud_project_kube_nodepool

### Terraform Configuration Files

```hcl
terraform {
  required_providers {
    ovh = {
      source  = "ovh/ovh"
      version = "~> 0.34.0"
    }
  }
  required_version = ">= 1.6.2"
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-7"
  flavor_name    = "b2-7"
  desired_nodes  = var.control_plane_b2_7_node_count
  anti_affinity  = false
  monthly_billed = false
  template {
    spec {
      unschedulable = false
      taints = [
        {
          effect = "PreferNoSchedule"
        }
      ]
    }
    metadata {
      annotations = {}
      finalizers  = ["ovhcloud.com/v1beta1", "ovhcloud.com/v1"]
      labels = {
        "node-restriction.kubernetes.io/flavor-range"   = "b2"
        "node-restriction.kubernetes.io/instance-type"  = "b2-7"
        "node-restriction.kubernetes.io/public-network" = "250M"
        "node-restriction.kubernetes.io/role"           = "controle-plane"
      }
    }
  }
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-30"
  flavor_name    = "b2-30"
  desired_nodes  = var.control_plane_b2_30_node_count
  anti_affinity  = false
  monthly_billed = false
  template {
    spec {
      unschedulable = false
      taints = [
        {
          effect = "PreferNoSchedule"
        }
      ]
    }
    metadata {
      annotations = {}
      finalizers  = ["ovhcloud.com/v1beta1", "ovhcloud.com/v1"]
      labels = {
        "node-restriction.kubernetes.io/flavor-range"   = "b2"
        "node-restriction.kubernetes.io/instance-type"  = "b2-30"
        "node-restriction.kubernetes.io/public-network" = "500M"
        "node-restriction.kubernetes.io/role"           = "controle-plane"
      }
    }
  }
}

```

### Expected Behavior

The resources are created without issues.

### Actual Behavior

The provider crashes

```
Stack trace from the terraform-provider-ovh_v0.34.0 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 116 [running]:
github.com/ovh/terraform-provider-ovh/ovh.loadNodelPoolTemplateFromResource({0x19d0480, 0xc0008b0440})
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:201 +0x9d4
github.com/ovh/terraform-provider-ovh/ovh.(*CloudProjectKubeNodePoolCreateOpts).FromResource(0xc000882320, 0x22?)
        github.com/ovh/terraform-provider-ovh/ovh/types_cloud_project_kube_nodepool.go:87 +0x24e
github.com/ovh/terraform-provider-ovh/ovh.resourceCloudProjectKubeNodePoolCreate(0x0?, {0x191a240?, 0xc0001aa510})
        github.com/ovh/terraform-provider-ovh/ovh/resource_cloud_project_kube_nodepool.go:242 +0x18c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x1b4e2c0?, {0x1b4e2c0?, 0xc0006cede0?}, 0xd?, {0x191a240?, 0xc0001aa510?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:695 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000472c40, {0x1b4e2c0, 0xc0006cede0}, 0xc00030d450, 0xc000015e80, {0x191a240, 0xc0001aa510})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:837 +0xa85
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0003e6e88, {0x1b4e2c0?, 0xc0006cecc0?}, 0xc000796460)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1021 +0xe8d
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000340000, {0x1b4e2c0?, 0xc0006ce360?}, 0xc0002682a0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x19d0320?, 0xc000340000}, {0x1b4e2c0, 0xc0006ce360}, 0xc000268230, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0001d61e0, {0x1b51db8, 0xc000007ba0}, 0xc00089a480, 0xc0004aae70, 0x20a08c0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xd33
google.golang.org/grpc.(*Server).handleStream(0xc0001d61e0, {0x1b51db8, 0xc000007ba0}, 0xc00089a480, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0xa36
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.53.0/server.go:963 +0x28a

Error: The terraform-provider-ovh_v0.34.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

2023-11-01T22:16:05.007+0100 [DEBUG] provider: plugin exited
```

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Additional context

Without specifying the template block the plugin does not crashes, example:

```hcl
resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_7_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-7"
  flavor_name    = "b2-7"
  desired_nodes  = var.control_plane_b2_7_node_count
  anti_affinity  = false
  monthly_billed = false
}

resource "ovh_cloud_project_kube_nodepool" "control_plane_b2_30_node" {
  service_name   = var.service_name
  kube_id        = data.terraform_remote_state.kube.outputs.cluster_id
  name           = "control-plane-b2-30"
  flavor_name    = "b2-30"
  desired_nodes  = var.control_plane_b2_30_node_count
  anti_affinity  = false
  monthly_billed = false
}
```

### PR DESCRIPTION
**Pull request by matprig**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #484  (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test: `make testacc TESTARGS="-run TestAccCloudProjectKubeNodePoolTaints"`

```

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or issues
- [x] I have added acceptance tests that prove my fix is effective or that my feature works
- [x] New and existing acceptance tests pass locally with my changes

### CODE REVIEW

File: ovh/resource_cloud_project_kube_nodepool_test.go [hunk 1]
```diff
@@ -81,6 +187,7 @@ resource "ovh_cloud_project_kube" "cluster" {
   version      = "%s"
 }
 
+
```
Comment (amstuta) [inline ovh/resource_cloud_project_kube_nodepool_test.go:190]:
```suggestion
```

File: ovh/resource_cloud_project_kube_nodepool_test.go [hunk 2]
```diff
@@ -73,6 +80,105 @@ func testSweepCloudProjectKubeNodePool(region string) error {
...
+          #key    = "t1"
+          value  = "tv1"
+        }
+      ]
+    }
+  }
+}
+`
+
+var testAccCloudProjectKubeNodePoolConfigForValueNoCrashTaints = `
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/resource_cloud_project_kube_nodepool_test.go]: yes i like it

File: website/docs/r/cloud_project_kube_nodepool.html.markdown [hunk 3]
```diff
@@ -81,6 +81,9 @@ The following arguments are supported:
         * `labels` - Labels to apply to each node
     * `spec` - Spec of each node in the pool
         * `taints` - Taints to apply to each node
+          * `effect` - mandatory possible values: NoExecute, NoSchedule, PreferNoSchedule
```
[the remark this answers is not in the document]
Comment (matprig) [reply website/docs/r/cloud_project_kube_nodepool.html.markdown:84]: i added the link to kubernetes documentation that describe these fields

File: ovh/types_cloud_project_kube_nodepool.go [hunk 4]
```diff
@@ -196,11 +196,16 @@ func loadNodelPoolTemplateFromResource(i interface{}) (*CloudProjectKubeNodePool
...
 					Key:    taint.(map[string]interface{})["key"].(string),
-					Value:  taint.(map[string]interface{})["value"].(string),
-				})
+				}
+				if taint.(map[string]interface{})["value"] != nil {
+					taintObject.Value = taint.(map[string]interface{})["value"].(string)
+				}
+
+				template.Spec.Taints = append(template.Spec.Taints, taintObject)
+
```
[the remark this answers is not in the document]
Comment (matprig) [reply ovh/types_cloud_project_kube_nodepool.go]: yes

**Review by amstuta — APPROVED** [review]

**Review by amstuta — APPROVED** [review]

**Pull request merged**